src/evm_context.rs

```diff
@@ -174,7 +174,16 @@ pub fn basefee(ssa: &mut SSATracker) -> SMTVariable {
 pub fn calldataload(index: SMTExpr, ssa: &mut SSATracker) -> SMTExpr {
     let calldata = context().calldata.smt_var(ssa);
     let arguments = (0..32)
-        .map(|i| smt::uf(calldata.clone(), vec![smt::bvadd(index.clone(), i)]))
+        .map(|i| {
+            // TODO this could benefit from let expressions.
+            // We are repeating the expression that computes the index.
```

Comment:
We could also just add a temporary variable?

Reply:
We don't have the mechanism to create a temporary variable here because we don't have access to the counter...

Reply:
we could move that to the ssa tracker